File: Poin B.cpp/Hardware.cpp
#include<string>
#include<iostream>
#include<string>
//Pemanggilan File Parent
#include"Product.cpp"

using namespace std;

class Hardware : public Product{
    //Deklarasi Atribut
    private :
        string brand;
        string model;

    public :
        
        //Constructor
        Hardware(){

        }
        
    
        //Get Set Methods
        void setBrand(string brand){
            this->brand = brand;
        }

        string getBrand(){
            return this->brand;
        }

        void setModel(string model){
            this->model = model;
        }

        string getModel(){
            return this->model;
        }
        
        //Output Method
        void output_Hardware(){
            cout << "Id Product                 : " << this->getId_Product() << endl;
            cout << "Price                      : " << this->getPrice() << endl;
            cout << "Brand                      : " << this->getBrand() << endl;
            cout << "Model                      : " << this->getModel() << endl;
        }
        
    
        //Destructor
        ~Hardware(){

        }

        
};

File: Poin A/Hardware.java
import javax.print.event.PrintEvent;

public class Hardware extends Product{
    
    //Deklarasi Atribut
    private String brand;
    private String model;
    
    //Constructor kosong
    public Hardware() {
    }
    
    //Constructor
    public Hardware(String brand, String model) {
        this.setBrand(brand);
        this.setModel(model);
    }
    
    //Get Set Methods
    public void setModel(String model) {
        this.model = model;
    }

    public String getModel() {
        return model;
    }

    public String getBrand() {
        return brand;
    }

    public void setBrand(String brand) {
        this.brand = brand;
    }
    
    //Output Methods
    public void output_hardware(){
        System.out.println("Id Product          : "+ this.getId_Product());
        System.out.println("Price               : "+ this.getPrice());
        System.out.println("Brand Hardware      : "+ this.getBrand());
        System.out.println("Model Hardware      : "+ this.getModel());

    }
    
    
    
}
